File: occasum-core/src/main/java/pw/stamina/occasum/properties/primitives/clamp/RangeDoubleClamp.java
package pw.stamina.occasum.properties.primitives.clamp;

final class RangeDoubleClamp implements DoubleClamp {
    private final double min, max;

    RangeDoubleClamp(double min, double max) {
        this.min = min;
        this.max = max;
    }

    @Override
    public double clamp(double value) {
        if (value < min) {
            return min;
        } else if (value > max) {
            return max;
        } else {
            return value;
        }
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/properties/primitives/clamp/RangeDoubleClamp.kt
package pw.stamina.occasum.properties.primitives.clamp

internal class RangeDoubleClamp(private val min: Double, private val max: Double) : DoubleClamp {

    override fun clamp(value: Double): Double {
        return when {
            value < min -> min
            value > max -> max
            else -> value
        }
    }
}
